src/lightning_app/utilities/proxies.py

```diff
@@ -74,7 +74,7 @@ def _send_data_to_caller_queue(work: "LightningWork", caller_queue: "BaseQueue",
 
     data.update({"state": work_state})
     logger.debug(f"Sending to {work.name}: {data}")
-    caller_queue.put(data)
+    caller_queue.put(deepcopy(data))
```

Comment:
So the issue was that `data` would change while the call to `put` was taking place, which is avoided now because data won't change in-place. Is that the correct interpretation?

Reply:
Yes, this is correct.